src/libutil/thread-pool.cc

```diff
@@ -110,10 +110,15 @@ void ThreadPool::doWork(bool mainThread)
                            propagate it. */
                         try {
                             std::rethrow_exception(exc);
+                        } catch (const Interrupted &) {
+                            // The interrupted state may be picked up multiple
```

Comment:
tiniest of tiny nits, feel free to ignore if you'd rather not wait for CI again :laughing: 

```suggestion
                            // The interrupted state may be picked up by multiple
```